### ISSUE TITLE
Measuring state variables

### ISSUE THREAD

**Issue by kaiolae** [Unity-Technologies/ml-agents#1880]
Hi!
Is there some way to have "state variables" that one could measure and use in the training? For instance, say my agent had some battery level that I wish to monitor, and that could affect it's decisions. How could I go about to implement such a variable, and how could it be exposed to the training algorithm in the gym interface?

Thanks!
-Kai

**Comment by xiaomaogy** [comment]
Hi @kaiolae , you could add that variable using the AddVectorObs api, please refer to this page for help. https://github.com/Unity-Technologies/ml-agents/blob/master/docs/Learning-Environment-Create-New.md

**Comment by kaiolae** [comment]
Thanks, @xiaomaogy ! I think I got it to work. It seems that to make this work, I had to also add some vector observations to the relevant brain, does that sound right?
My agent now has both vector and visual observations. Is that supported?
Finally, I was wondering, how can I access these vector observations in the gym-interface? When I call gym's step method, I get back the visual observation, but I can't seem to get the vector observation back.
Thanks again!
